crates/hyperswitch_connectors/src/connectors/fiuu.rs

```diff
@@ -136,17 +138,48 @@ impl ConnectorCommon for Fiuu {
 
         Ok(ErrorResponse {
             status_code: res.status_code,
-            code: response.code,
-            message: response.message,
-            reason: response.reason,
+            code: response.error_code.clone(),
+            message: response.error_desc.clone(),
+            reason: Some(response.error_desc.clone()),
             attempt_status: None,
             connector_transaction_id: None,
         })
     }
 }
+pub fn build_form_from_struct<T: Serialize>(data: T) -> Result<Form, common_errors::ParsingError> {
+    let mut form = Form::new();
+    let serialized = serde_json::to_value(&data)
+        .map_err(|_| common_errors::ParsingError::EncodeError("json-value"))?;
+    let serialized_object = serialized
+        .as_object()
+        .ok_or(common_errors::ParsingError::EncodeError("Expected object"))?;
+
+    for (key, values) in serialized_object {
+        let value = match values {
+            Value::String(s) => s.clone(),
+            Value::Number(n) => n.to_string(),
+            Value::Bool(b) => b.to_string(),
+            _ => "".to_string(),
```

Comment:
We'd end up serializing arrays and nested object with an empty string value. I'm sure we shouldn't be doing that?